### PR TITLE
Upload comments

### PR DESCRIPTION
**Pull request by anushadatta**
Added functionality to upload comments (text and timestamp) to the server by making a POST request through a simple form. After the comment is added, the user is redirected to the same page and is able to view the updated comments section. 

Also, removed the hard coded comments data from the server. Added logic to handle the event in which there is no comments history (0 comments).

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java [hunk 1]
```diff
@@ -44,6 +41,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    // Upload comment 
+    uploadComment(date, text);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/blog.html");
+  }
+
+  private void uploadComment(Date date, String text) {
+
+      BlogComment c = new BlogComment(date, text);
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java:59]: We need better variable names :D

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -40,27 +40,34 @@ function getBlogComments() {
...
 
-                var comment = document.createElement("div");
-                comment.classList.add("comment-card");
-                comment.innerHTML = createCommentElement(commentsList[i]);
-
-                commentsDiv.appendChild(comment);
+                commentsDiv.appendChild(noComments);
             }
+
+            else {
```
Comment (rashikaranpuria) [inline portfolio/src/main/webapp/script.js:50]: Blank line between if and else, It might be better to have else on same line as closing brace

**Review by rashikaranpuria — CHANGES_REQUESTED** [review]

File: portfolio/pom.xml [hunk 3]
```diff
@@ -29,6 +29,12 @@
       <artifactId>gson</artifactId>
       <version>2.8.6</version>
     </dependency>
+
+    <dependency>
+    <groupId>com.google.appengine</groupId>
```
[the remark this answers is not in the document]
Comment (anushadatta) [reply portfolio/pom.xml:34]: Whoops, okay!

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java [hunk 4]
```diff
@@ -44,6 +41,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+      commentsList.add(c);
+
+      return;
```
[the remark this answers is not in the document]
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java:62]: Oh right, sure I'll remove that then.

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java [hunk 5]
```diff
@@ -27,14 +27,11 @@
 @WebServlet("/comments")
 public final class BlogCommentsServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Comments data
     ArrayList<BlogComment> commentsList = new ArrayList<BlogComment>();
```
[the remark this answers is not in the document]
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java:31]: Got it, I'll make that change.

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java (hunk 1 above)
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java:59]: I hear you! I'll switch to a more verbose variable name.

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (anushadatta) [reply portfolio/src/main/webapp/script.js:50]: You're right, that would work better. I'll make that change. 👍

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java]: For some reason changing it to private was causing compilation/build errors - so I just changed it to protected.

**Review by rashikaranpuria — COMMENTED** [review]
LGTM

**Pull request merged**